src/helpers/readme-helpers-trait.php

```diff
@@ -0,0 +1,187 @@
+			if ( $found['isDeprecatedLicenseId'] ) {
+				$response->deprecated[ $response->provided ] = $found;
+			} else {
+				$response->live[ $response->provided ] = $found;
+			}
+		}
+
+		if ( ! $found ) {
+			foreach ( $this->license_data as $license => $details ) {
+				if ( in_array( $response->provided, $details['names'], true ) || in_array( $response->provided, $details['ids'], true ) ) {
+					if ( $this->license_data[ $license ]['isDeprecatedLicenseId'] ) {
+						$response->deprecated[ $license ] = $details;
+					} else {
+						$response->live[ $license ] = $details;
+					}
+				}
+			}
+		}
+
+		return $this->get_response_message( $response );
+	}
+
+	/**
+	 * Get license match messages.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param Object $response Object containing license critera and match information.
+	 *
+	 * @return Object $response Object containing license critera and match information.
+	 */
+	public function get_response_message( $response ) {
+
+		// Non-deprecated license matches found - check here.  Names can match deprecated licenses as well.
+		if ( ! empty( $response->live ) ) {
+
+			// Single live license match.
+			if ( count( $response->live ) === 1 ) {
+				$details = current( $response->live );
+
+				// Set the ID for easier lookup in other checks.
+				$response->id = $details['licenseId'];
+
+				// Provided a SPDX name that matched.
+				if ( $response->provided === $details['name'] ) {
+					$response->status = 'warning';
+					/* translators 1: a SPDX license name. 2: the recommended SPDX ID to use instead. */
+					$response->message = sprintf( 'Found a valid SPDX name, %1$s, but it is better to use the SPDX ID: %2$s', $response->provided, $details['licenseId'] );
+				} elseif ( $response->provided === $details['licenseId'] ) { // A Valid SPDX ID was found, no message required.
+					$response->status  = 'success';
+					$response->message = null;
+				} else { // A single match was found for FSF critera.
+					$response->status = 'warning';
+					/* translators 1: a SPDX license name. 2: the recommended SPDX ID to use instead. */
+					$response->message = sprintf( 'Found valid license information based on FSF naming: %1$s, but it is better to use the SPDX ID: %2$s', $response->provided, $details['licenseId'] );
+				}
+			} else { // Multiple matches returned, so it's FSF provided critera.
+				$matches          = array_keys( $response->live );
+				$response->status = 'error';
+				/* translators %s: listing of license IDs matched. */
+				$response->message = sprintf( 'Found multiple records matching these licenses: %s, it\'s required to use a single SPDX Idenitfier!', implode( ', ', $matches ) );
+			}
+		} elseif ( ! empty( $response->deprecated ) ) { // Deprecated match found.
+			$response->status  = 'error';
+			$response->message = sprintf( __( 'The license identification provided, $s, indicates a deprecated license!  Please use a valid SPDX Identifier!', 'theme-sniffer' ), $response->provided );
```

Comment:
Can we use `esc_html__()` here instead of just `__()`? It's not a security issue, but I just like when everything is properly escaped 🙂